### ISSUE TITLE
Pop!_OS: text input hangs after certain combinations of input or moving/resizing windows with use_ime=true

### ISSUE THREAD

**Issue by MostHated** [wez/wezterm#1628]
### What Operating System(s) are you seeing this problem on?

Linux X11

<details>
<summary>System Details</summary>

```sh
mosthated@pop-os
----------------
OS: Pop!_OS 21.10 x86_64
Host: X570 AORUS ELITE -CF
Kernel: 5.15.15-76051515-generic
Uptime: 2 days, 23 hours, 38 mins
Packages: 4016 (dpkg), 193 (brew), 81 (flatpak), 13 (snap)
Shell: zsh 5.8
Resolution: 1920x1080, 2560x1080, 1920x1080
DE: GNOME 40.5
WM: Mutter
WM Theme: Materia-dark-compact
Theme: Materia-dark-compact [GTK2/3]
Icons: Flatery-Orange-Dark [GTK2/3]
Terminal: WezTerm
CPU: AMD Ryzen 9 3950X (32) @ 3.500GHz
GPU: NVIDIA GeForce RTX 2070
Memory: 44336MiB / 64314MiB
```

</details>

### WezTerm version

20220208-064631-dbf190d4

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

Upon testing the nightly AppImage I discovered that if I split the pane, everything is good for a few seconds, but after about 5-10 seconds, the terminal no longer accepts any keyboard input. Mouse input still works, ex I can click/right-click the tab menus, but can't type anything in either of the pane splits or the new tab if I had created one.

### To Reproduce

Open nightly AppImage terminal, split the pane (I had only tested splitting left -> right) and then I resized the box a few times and placed it where I wanted it, then went to type into the right pane and nothing happened. I tried the left pane and had the same thing.

### Configuration

```lua
local wezterm = require 'wezterm';
return {
  font = wezterm.font("CaskaydiaCove NF"),
  color_scheme = "Batman",
}
```

### Expected Behavior

Text input would continue to be accepted at all times.

### Logs

Can't pull up the log once this occurs, unfortunately.

### Anything else?

Action shot

![](https://i.imgur.com/wTmFWSy.gif)

**Comment by MostHated** [comment]
P.S. This terminal is great. It looks and feels so snappy and... crisp? Even more so than Alacritty, which I imagine was not trivial. This might become my new daily driver over it. :+1:

**Comment by wez** [comment]
Can you start wezterm from another terminal and reproduce this?  It should log something to the stderr where it was originally logged and I think that will give us a clue about what is going on

**Comment by MostHated** [comment]
Unfortunately, it was not very helpful. : /

<details>
<summary>Details</summary>

```sh
~/Downloads via ☕ v1.8.0
zsh  ./WezTerm-nightly-Ubuntu18.04.AppImage                                                                                             mosthated@pop-os
 2022-02-09T15:36:37.560Z WARN  wezterm_gui > while attempting to ask existing instance to spawn: Running GUI is a different executable from us, will start a new one
 2022-02-09T15:36:37.584Z INFO  wezterm_mux_server_impl::local > setting up /run/user/1000/wezterm/gui-sock-2855942
 2022-02-09T15:36:37.700Z INFO  wezterm_gui::termwindow        > OpenGL initialized! NVIDIA GeForce RTX 2070/PCIe/SSE2 4.6.0 NVIDIA 470.86 is_context_loss_possible=false wezterm version: 20220208-064631-dbf190d4

~/Downloads via ☕ v1.8.0 took 27s
zsh  ./WezTerm-nightly-Ubuntu18.04.AppImage                                                                                             mosthated@pop-os
 2022-02-09T15:37:07.240Z WARN  wezterm_gui > while attempting to ask existing instance to spawn: Running GUI is a different executable from us, will start a new one
 2022-02-09T15:37:07.265Z INFO  wezterm_mux_server_impl::local > setting up /run/user/1000/wezterm/gui-sock-2859608
 2022-02-09T15:37:07.391Z INFO  wezterm_gui::termwindow        > OpenGL initialized! NVIDIA GeForce RTX 2070/PCIe/SSE2 4.6.0 NVIDIA 470.86 is_context_loss_possible=false wezterm version: 20220208-064631-dbf190d4

~/Downloads via ☕ v1.8.0 took 14s
zsh  ./WezTerm-nightly-Ubuntu18.04.AppImage                                                                                             mosthated@pop-os
 2022-02-09T15:37:24.260Z WARN  wezterm_gui > while attempting to ask existing instance to spawn: Running GUI is a different executable from us, will start a new one
 2022-02-09T15:37:24.284Z INFO  wezterm_mux_server_impl::local > setting up /run/user/1000/wezterm/gui-sock-2861904
 2022-02-09T15:37:24.406Z INFO  wezterm_gui::termwindow        > OpenGL initialized! NVIDIA GeForce RTX 2070/PCIe/SSE2 4.6.0 NVIDIA 470.86 is_context_loss_possible=false wezterm version: 20220208-064631-dbf190d4
```

</details>


![](https://i.imgur.com/hBP2oVj.gif)

**Comment by wez** [comment]
Try: `WEZTERM_LOG=window=trace,info wezterm start --always-new-process` to get some additional logging

**Comment by wez** [comment]
Please also try alt-tabbing away and back to the window to see if the input is recognized after doing that

**Comment by MostHated** [comment]
Ok, here is what I have. 


Here is a clip of the run: 

https://i.imgur.com/rCsKicA.mp4

Attached is the output log from that session.

[wezterm_output.log](https://github.com/wez/wezterm/files/8034382/wezterm_output.log)

The last successful key input was on line 1009 : 2022-02-09T15:50:09.602Z  and roughly 00:24-25 seconds into the clip

I also went back after I saw your comment and tried it again, seeing if alt+tab made a difference, but unfortunately it did not. What I do notice, though, is that when starting the terminal like this, the issue is harder to reproduce. Still reproducible, but I have to work for it, lol.

**Comment by wez** [comment]
Can you try setting `use_ime = false`?

**Comment by MostHated** [comment]
This is with  use_ime = false in the config:
```sh
zsh  echo $XMODIFIERS   
@im=ibus
```

**Comment by wez** [comment]
Fixed by 1485cf344cebcb0a73b69b498813d20da659e9fb!

It typically takes about an hour before fixes are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out.

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them.

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.
